src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,2383 @@
+			SystemSensibleHeatingEnergy = 0;
+			SystemLatentHeatingRate = 0;
+			SystemLatentHeatingEnergy = 0;
+			SupplyFanElectricPower = 0;	   
+			SupplyFanElectricEnergy = 0;    
+			SecondaryFuelConsumptionRate = 0;
+			SecondaryFuelConsumption = 0;   
+			ThirdFuelConsumptionRate = 0;   
+			ThirdFuelConsumption = 0;	   
+			WaterConsumptionRate = 0;	   
+			WaterConsumption = 0;		   
+			ExternalStaticPressure = 0;	 
+			QLatentZoneOut = 0;
+			QSensZoneOut = 0;
+			//debug
+			//RawHR = 0;
+		}
+
+		void Model::InitializeModelParams()
+		{
+			// SUBROUTINE INFORMATION:
+			//       AUTHOR         Spencer Maxwell Dutton
+			//       DATE WRITTEN   October 2017 
+			//       MODIFIED       
+			//       RE-ENGINEERED  na
+
+			// PURPOSE OF THIS SUBROUTINE:
+			// Reset calculation values 
+
+			// METHODOLOGY EMPLOYED:
+			// 
+
+			// REFERENCES:
+			// na
+
+			// Using/Aliasing
+			ResetOutputs();
+			PrimaryMode = 0;
+			PrimaryModeRuntimeFraction = 0;
+			optimal_EnvCondMet = false;
+			Tsa = 0;
+			//reset the power use to a high value, this is replaced during the calculation keeping the "best" setting. 
+		
+			RunningPeakCapacity_EnvCondMet = false;
+			Settings.clear();
+		}
+
+		Model::~Model()                 
+		{
+		}
+
+		CMode::~CMode()  
+		{
+	
+		}
+
+
+		void Model::Initialize(int ZoneNumber)
+		{
+			// SUBROUTINE INFORMATION:
+			//       AUTHOR         Spencer Maxwell Dutton
+			//       DATE WRITTEN   October 2017 
+			//       MODIFIED       
+			//       RE-ENGINEERED  na
+
+			// PURPOSE OF THIS SUBROUTINE:
+			// Specify solution space resolution, and populate the solution spaces in each mode
+
+			// METHODOLOGY EMPLOYED:
+			// Solution spaces are the matrices of possible settings settings (combination of OSA fraction and supply air mass flow rate) 
+			// This method calls the GenerateSolutionSpace for each of the modes defined in the idf. 
+			// REFERENCES:
+			// na
+
+			// Using/Aliasing
+
+			ZoneNum = ZoneNumber;
+			if (Initialized){
+				return;
+			}
+			Initialized = true;
+			// this variable, at this point hard coded, sets the fidelity of the matrix of possible OSAF and Msa combinations. 
+			ResolutionMsa = 0.2; //msa/msaRATED
+			ResolutionOSA = 0.2; //OSAF as absolute fraction (not %)
+		
+			//Iterate through modes of operation generating a matrix of OSAF and Msa to test in the algorithm.
+			std::vector<CMode>::iterator iterModes;
+			for (iterModes = OperatingModes.begin(); iterModes != OperatingModes.end(); ++iterModes) {
+				(*iterModes).GenerateSolutionSpace(ResolutionMsa, ResolutionOSA);
+			}
+
+			Initialized = true;
+		}
+
+
+		Real64 Model::Part_press(Real64 P, Real64 W)
+		{
+			// Function to compute partial vapor pressure in [kPa]
+			// From page 6.9 equation 38 in ASHRAE Fundamentals handbook (2005)
+			//   P = ambient pressure [kPa]
+			//   W = humidity ratio [kg/kg dry air]
+
+			return (P * W / (0.62198 + W));
+		}
+		
+		Real64 Model::Sat_press(Real64 Tdb)

```

Comment:
We should already have this in Psychrometrics.hh, we won't want to have this repeated here.

Reply:
done